app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,57 @@
+class CartsController < ApplicationController
+  before_action :load_product_with_quantity, only: %i(create update)
+  def index
+    @products_in_cart = {}
+
+    current_carts.each do |id, quantity|
+      product = Product.find_by(id: id)
+      @products_in_cart[product] = quantity.to_i
+    end
+  end
+
+  def create
+    if @quantity.positive? && @quantity < @product.quantity
+      add_to_cart params[:id], params[:quantity]
+      flash[:success] = t "success"
+    else
+      flash[:warning] = t "fails"
+    end
+    redirect_to carts_path
+  end
+
+  def update
+    update_quantity_cart params[:id], params[:quantity]
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  def destroy
+    delete_item_cart params[:id]
+
+    redirect_to carts_path
+  end
+
+  private
+  def add_to_cart id, quantity
+    current_carts[id] = if current_carts.key?(id)
+                          current_carts[id] + quantity.to_i
+                        else
+                          quantity.to_i
+                        end
```

Comment:
Trên này cũng áp dụng toán tử 3 ngôi được không em

Reply:
dùng toán tử 3 ngôi ở đây nó bị dòng quá dài, nếu xuống dòng thì vẫn bị rubocop check fails vì không được break line trong toán tử 3 ngôi